packages/infra/infra-core/src/lib/patterns/ecr-sync/ecr-sync.ts

```diff
@@ -0,0 +1,207 @@
+
+  /**
+   * A prefix for all ECR repository names.
+   *
+   * @default Empty.
+   */
+  readonly repoPrefix?: string;
+
+  /**
+   * Optional. Schedule when images should be synchronized.
+   *
+   * @default is once a day.
+   */
+  readonly schedule?: evt.Schedule;
+
+  /**
+   * Optional. Bash script injection for the docker image processing phase,
+   * in order to log in to Dockerhub or do other initialization.
+   *
+   * @default Empty.
+   */
+  readonly initScript?: string;
+}
+
+/**
+ * Construct to sync Docker images from DockerHub into ECR Repos.
+ */
+export class EcrSync extends Construct {
+
+  private ecrRepos: ecr.Repository[] = [];
+
+  constructor(scope: Construct, id: string, props: EcrSyncProps) {
+    super(scope, id);
+
+    const artifactsBucket = new s3.Bucket(this, 'ArtifactBucket', {
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      enforceSSL: true,
+      versioned: true,
+    });
+
+    const buildRole = new iam.Role(this, 'buildRole', {
+      assumedBy: new iam.ServicePrincipal('codebuild.amazonaws.com'),
+    });
+
+    const lambaFile = `${path.resolve(__dirname)}/lambda/get-image-tags-handler`;
+    const entry = lambaFile + (fs.existsSync(`${lambaFile}.ts`) ? '.ts' : '.js');
+
+    const lambda = new lnjs.NodejsFunction(this, 'lambda', {
+      functionName: props.getImageTagsFunctionName,
+      entry: entry,
+      runtime: lmbd.Runtime.NODEJS_18_X,
+      timeout: cdk.Duration.minutes(10),
+      logRetention: logs.RetentionDays.ONE_WEEK,
+      memorySize: 256,
+      environment: {
+        AWS_ACCOUNT_ID: cdk.Stack.of(this).account,
+        REGION: cdk.Stack.of(this).region,
+        IMAGES: JSON.stringify(props.dockerImages),
+        REPO_PREFIX: props.repoPrefix ?? '',
+        BUCKET_NAME: artifactsBucket.bucketName,
+      },
+    });
+    artifactsBucket.grantPut(lambda);
+
+    new evt.Rule(this, 'ScheduleGetImageTags', {
+      targets: [new trgt.LambdaFunction(lambda)],
+      schedule: props.schedule ?? evt.Schedule.rate(cdk.Duration.days(1)),
+    });
+
+    props.dockerImages.forEach(element => {
+      const repoName = (props.repoPrefix) ? `${props.repoPrefix}/${element.imageName}` : element.imageName;
+      const repo = new ecr.Repository(this, element.imageName, {
+        repositoryName: repoName,
+      });
+      repo.grantPullPush(buildRole);
+
+      lambda.addToRolePolicy(new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ['ecr:ListImages'],
+        resources: [repo.repositoryArn],
+      }));
+
+      if (props.lifcecyleRule !== undefined) {
+        repo.addLifecycleRule(props.lifcecyleRule);
+      }
+
+      this.ecrRepos.push(repo);
+    });
+
+    // Get triggered by s3
+    const triggerStageArtifact = new cp.Artifact();
+    const triggerAction = new cpa.S3SourceAction({
+      actionName: 'S3Source',
+      bucket: artifactsBucket,
+      bucketKey: 'images.zip',
```

Comment:
How about moving this filename to some kind of a constant value? I see it's also referenced inside `packages/infra/infra-core/src/lib/patterns/ecr-sync/lambda/get-image-tags-handler.ts`

Reply:
Actually, that is a library that I copied and pasted into repository :D 
Latest version was quite old, and used CDK v1, but on master I found a rewrite to v2 which hasn't been released yet.

Reply:
And I think the problem with this is that `get-image-tags-handler.ts` is a lambda function so all it's code has to be in the `lambda` directory. 
I wouldn't feel confident with CDK stack code importing any code from inside the lambda function code. It just feels wrong, even if for avoiding such issue. But I think we can use env variable in functions for that so help this issue.